client/a8c-for-agencies/sections/sites/site-actions/use-site-actions.tsx

```diff
@@ -0,0 +1,467 @@
+				isEnabled: isWPCOMSite && ! isUrlOnly,
+			},
+			{
+				name: translate( 'Issue new license' ),
+				onClick: () => {
+					page( A4A_MARKETPLACE_LINK );
+					handleClickMenuItem( 'issue_license' );
+				},
+				isExternalLink: false,
+				isEnabled: ! siteError && ! isWPCOMSite && ! isUrlOnly,
+			},
+			{
+				name: translate( 'View activity' ),
+				href: isWPCOMSite ? `https://wordpress.com/activity-log/${ blog_id }` : null,
+				onClick: () => {
+					if ( ! isWPCOMSite ) {
+						setDataViewsState( ( prevState: DataViewsState ) => ( {
+							...prevState,
+							selectedItem: site?.value,
+							type: DATAVIEWS_LIST,
+						} ) );
+						setSelectedSiteFeature( JETPACK_ACTIVITY_ID );
+					}
+					handleClickMenuItem( 'view_activity' );
+				},
+				isExternalLink: isWPCOMSite,
+				isEnabled: ! siteError && ! isUrlOnly,
+			},
+			{
+				name: translate( 'Copy this site' ),
+				href: isWPCOMSite ? `https://wordpress.com/backup/${ siteSlug }/clone` : null,
+				onClick: () => {
+					if ( ! isWPCOMSite ) {
+						setDataViewsState( ( prevState: DataViewsState ) => ( {
+							...prevState,
+							selectedItem: site?.value,
+							type: DATAVIEWS_LIST,
+						} ) );
+						setSelectedSiteFeature( JETPACK_BACKUP_ID );
+					}
+					handleClickMenuItem( 'clone_site' );
+				},
+				isExternalLink: isWPCOMSite,
+				isEnabled: has_backup && ! isUrlOnly,
+			},
+			{
+				name: translate( 'Site settings' ),
+				href: `https://wordpress.com/settings/general/${ blog_id }`,
+				onClick: () => handleClickMenuItem( 'site_settings' ),
+				isExternalLink: true,
+				isEnabled: isWPCOMSite && ! isUrlOnly,
+			},
+			{
+				name: translate( 'View site' ),
+				href: urlWithScheme,
+				onClick: () => handleClickMenuItem( 'view_site' ),
+				isExternalLink: true,
+				isEnabled: true,
+			},
+			{
+				name: translate( 'Visit WP Admin' ),
+				href: `${ urlWithScheme }/wp-admin`,
+				onClick: () => handleClickMenuItem( 'visit_wp_admin' ),
+				isExternalLink: true,
+				isEnabled: true && ! isUrlOnly,
+			},
+			{
+				name: translate( 'Remove site' ),
+				onClick: () => handleClickMenuItem( 'remove_site' ),
+				icon: 'trash',
+				className: 'is-error',
+				isEnabled: canRemove,
+			},
+			{
+				name: translate( 'Delete site' ),
+				onClick: () => handleClickMenuItem( 'delete_site' ),
+				icon: 'trash',
+				className: 'is-error',
+				isEnabled: canDelete,
+			},
+		];
+	}, [
+		canDelete,
+		canRemove,
+		dispatch,
+		isDevSite,
+		isLargeScreen,
+		isWPCOMSimpleSite,
+		isWPCOMSite,
+		onSelect,
+		setDataViewsState,
+		setSelectedSiteFeature,
+		site?.value,
+		siteError,
+		siteValue,
+		translate,
+	] );
+}
+
+type SiteActions = {
+	isLargeScreen: boolean;
+	onRefetchSite?: () => Promise< unknown >;
+	setDataViewsState: ( callback: ( prevState: DataViewsState ) => DataViewsState ) => void;
+	setSelectedSiteFeature: ( siteFeature: string | undefined ) => void;
+};
+
+const createRemoveSiteActionModal =
```

Comment:
I think this component can be extracted to its own file, and so we can rename back the file to `.ts`. GitHub hasn't been good at detecting the renaming and what are the actual changes.

Reply:
Done at https://github.com/Automattic/wp-calypso/pull/98276/commits/e4a97574ad95b41fcb7b391e36c5637a1f1e9f3f